### ISSUE TITLE
Consider a JS build tooling

### ISSUE THREAD

**Issue by xymbol** [cherta/js.uy#15]
A state-of-the-art website build using JS tooling would probably translate into one more opportunity to demonstrate the local community's skills. Then, it would better integrate with other à la mode JS tools: test coverage, code quality, other badges, etc.

**Comment by cherta** [comment]
Before using `hugo` I evaluated [some JS static site generators](https://www.staticgen.com/) and frankly, most of them were [too basic](http://www.metalsmith.io/) or [too](https://www.gitbook.com/) [generic](http://brunch.io/). 

I never considered rolling out our own tool for generating the site because I was more focused on creating the initial content and onboarding people. 

Considering the initial goals were accomplished now it's maybe a good time to do that and an opportunity not only to show what we do but to have a better "deployment" strategy, currently it involves pushing the generated HTML files into the `/docs` folder.

**Comment by xymbol** [comment]
@cherta Great approach, I'm all for MVPs and "release early, release often". Just wanted to document the question as an issue, at least for later reference. It will come up, I'm sure.

- Is the deployment documented?
- Does it need to be static?
- Any features we might want?

I would like to see something like a [twelve-factor](https://12factor.net/) app for the website so that it can be easily worked on and deployed.

**Comment by cherta** [comment]
>Is the deployment documented?

Yes, it's on the [README](https://github.com/js-uy/js.uy#how-to-build-this-site) and on the [CONTRIBUTING](https://github.com/js-uy/js.uy/blob/master/CONTRIBUTING.md) guidelines.

>Does it need to be static?

No, it can be anything that delivers a working website. I do think static sites enable ease of use and less moving parts, as a drawback, you probably can do less.

>Any features we might want?

The only feature that I can say it's a must for me is lowering the entry barrier to adding contributions to the site or the community in general.

* Better deployment/publishing: A user edits a file or posts pushes the code (or hits publish if it's a hosted app) and the site gets updated, no more running `hugo` or any other similar command prior pushing the code.
* Give ownership: I like the way [tettra](https://tettra.co/) treats they wiki pages it generates, it adds the slack avatar of the page collaborators

I have other ideas added in the [ideas column](https://github.com/js-uy/js.uy/projects/1) of the website project.

**Comment by MatiasVerdier** [comment]
@cherta 

> Better deployment/publishing: A user edits a file or posts pushes the code (or hits publish if it's a hosted app) and the site gets updated, no more running hugo or any other similar command prior pushing the code.

With [Netlify](https://www.netlify.com) you can achieve that very easily. Just define your build command and it's done, every time a change it's pushed to the repo a build is triggered and the site is deployed with the last updates.

Also you can check [NetlifyCMS](https://www.netlifycms.org/) for editing content right in the live site
